### DISCUSSION TITLE
DivideByZero exception when creating sound effects

### DISCUSSION THREAD

**Discussion by george7378** [MonoGame/MonoGame#8507] [Audio]
Hey everyone - I wonder if anyone knows what's causing this crash when creating sound effects?

`Exception android.runtime.JavaProxyThrowable: [System.DivideByZeroException]: Arg_DivideByZero
  at Microsoft.Xna.Framework.Audio.OALSoundBuffer.BindDataBuffer
  at Microsoft.Xna.Framework.Audio.SoundEffect.PlatformInitializePcm
  at Microsoft.Xna.Framework.Audio.SoundEffect..ctor
  at Microsoft.Xna.Framework.Content.SoundEffectReader.Read`

I've seen it come in crash reports on some android devices, but never seen it myself in testing.

Is there a good way to check for device compatibility, or to handle it - just try/catch when creating effects, and proceed with no sound?

**Comment by dellis1972** [comment]
You might be hitting an issue with ADPCM found effects. 
It was recently fixed via https://github.com/MonoGame/MonoGame/commit/2dd7b76bec5c02e9d30a900fb33f8914e4e3e9f8#diff-900cdc6f3184a1d1528a8dd41182468714651d59ffb740fd99b7d49697b70395R70

that said if it works in alot of places it might be device related.
